R/router.R

```diff
@@ -131,7 +137,14 @@ create_router_callback <- function(root, routes) {
         parsed <- parse_url_path(cleaned_url)
         parsed$path <- ifelse(parsed$path == "", root, parsed$path)
 
-        if (!valid_path(routes, parsed$path)) {
+        path_validation <- if (is.null(routes)) {
+          log_msg("Valid paths:", input$routes)
+          !is.null(input$routes) && !(parsed$path %in% c(input$routes, "404"))
+        } else {
+          !valid_path(routes, parsed$path)
+        }
```

Comment:
Hey I didn't know one can save the result of an if statement to a variable. Cool!